test/DropdownMenuSpec.js

```diff
@@ -33,4 +32,32 @@ describe('<Dropdown.Menu>', () => {
       </DropdownMenu>,
     ).assertSingle('.dropdown-menu-right');
   });
+
+  // it.only('warns about bad refs', () => {
+  //   class Parent extends React.Component {
+  //     componentDidCatch() {}
+
+  //     render() {
+  //       return this.props.children;
+  //     }
+  //   }
+
+  //   class Menu extends React.Component {
+  //     render() {
+  //       const { show: _, alignRight: _1, close: _2, ...props } = this.props;
+
+  //       return <div {...props} />;
+  //     }
+  //   }
+
+  //   expect(() =>
+  //     mount(
+  //       <Parent>
+  //         <DropdownMenu show as={Menu}>
+  //           <DropdownItem>Item</DropdownItem>
+  //         </DropdownMenu>
+  //       </Parent>,
+  //     ),
+  //   ).to.throw();
```

Comment:
IDK how to catch this error, anyone got ideas?

Reply:
Can we use `shouldWarn`?

Do we really want this to hard throw?

Reply:
In the worst case, you could put an error boundary around this guy, and assert on the error boundary having been hit

Reply:
this has an error boundary that doesn't seem to be hit. The error also still gets thrown crashing the test, which is confusing.

Reply:
it's gonna hard error somewhere in Popper if we only warn

Reply:
What happens if you try either using react test utils directly, or else doing shallow?

Reply:
shallow doesn't work with hooks really, and i've not had much luck doing anything directly. you wanna take a shot? I may just be dense